swiftlang/llvm-project
# Revert "[Coverage] Store compilation dir separately in coverage mapping"

This reverts commit 5fbd1a333aa1a0b70903d036b98ea56c51ae5224.

See: rdar://82543962 (Revert coverage changes from clang-1316 to make
coverage format in clang-1316 identical to clang-1300)

Testing:
  * check-profile passed before and after this revert.
  * check-clang reported 4 failing tests prior to this revert; that
    remains unchanged. Driver/clang_f_opts.c required a workaround. The
    coverage-compilation-dir.c test was xfailed.
  * check-llvm passed before and after this revert. The
    llvm-cov/compilation_dir.c test is now xfailed. Coverage unit tests
    pass with the exception of non_code_region_counters, which was
    disabled because https://reviews.llvm.org/D101780 was folded into
    this revert.

 Conflicts:
	clang/include/clang/Basic/CodeGenOptions.h
	clang/include/clang/Driver/Options.td
	clang/lib/CodeGen/CoverageMappingGen.cpp
	clang/lib/Driver/ToolChains/Clang.cpp
	clang/test/Profile/profile-prefix-map.c
	llvm/include/llvm/ProfileData/Coverage/CoverageMappingReader.h
	llvm/lib/ProfileData/Coverage/CoverageMappingReader.cpp
	llvm/unittests/ProfileData/CoverageMappingTest.cpp
